### DISCUSSION TITLE
Regarding web-eye-tracker project

### DISCUSSION THREAD

**Discussion by sajontahsen** [ruxailab/RUXAILAB#331] [Q&A] [answered]
Hello, I'm interested in working on 'Eye Tracking Algorithm Optimization' for GSoC'24. Looking for some directions on the following:
- I wanted to work on [this](https://github.com/ruxailab/web-eye-tracker/issues/6) and [this](https://github.com/ruxailab/web-eye-tracker/issues/9) issue in the web-eye-tracker repo, to also get familiar with the codebase as well. I noiced the endpoints in main and develop branch are very much different, so should i fork the develop branch and base my work on that, or stick to main? 
- I noticed that the model used in `gaze_tracker.py` is LinearRegression, and found some more experiments [here](https://github.com/ruxailab/web-eye-tracker/blob/c8494512470aeee6c33688e5a633765432c2b979/app/services/data_exploration.ipynb) . I'd love to train and evaluate some models on my end and explore other possibilities for improvement. Is the training data publicly available? Any direction regarding the data would be very much appreciated. 

I found the web-eye-tracker project to be particularly exciting. Would love to discuss more on how i can contribute effectively. 

Best regards,
Tahsen Islam Sajon
[Resume](https://drive.google.com/file/d/1nUR4WYFnh4pY0yUSeteKR8PqZHS_9xgh/view?usp=drivesdk) | [Email](mailto:sajon.tahsen@gmail.com) | [LinkedIn](https://www.linkedin.com/in/tahsen-islam-sajon)

**Comment by KarinePistili** [comment, marked as answer]
Hello @LostSputnik , thank you for the interest. Something that would interest us a lot would be to experiment with different algorithms, and also a real-time proof of concept.

The eye tracker project has two repositories that work together. The frontend version captures the eye movements and does the callibration, while the backend contains the linear regression model.

You will need to run them together, so that's how you would get the data you can use to train your datasets.  I'll leave this discussion opened so we can keep talking here.

**Reply by sajontahsen** [reply]
Thanks for the answer @KarinePistili. I went through the model code trying to figure out how a real-time POC can be set up. 
However, I've been having a lot of trouble setting up the project. I've opened a issue [here](https://github.com/ruxailab/web-eye-tracker/issues/12). 
I'd really like to get to work on the POC, so any help would be really appreciated. Thanks

**Comment by sushant4612** [comment]
I am facing difficulty in setting up the project locally
it's showing the following error

https://github.com/ruxailab/RUXAILAB/assets/109458485/679bd8c9-3b05-4fac-a379-787251634817

**Reply by sushant4612** [reply]
@KarinePistili  could you please help me out with this?

**Reply by KarinePistili** [reply]
Hello @sushant4612 ! I believe it could be something related to version incompatibility.

 I'll ping here @hvini and @Natanco1 who can have some insight on what is happening to your installation and help here on the discussion ;)

**Comment by sajontahsen** [comment]
Hello @KarinePistili. Some updates from me: I was able to get the project up and running, and am trying to better understand how the real-time features should be integrated. I'm linking the discussion [here](https://github.com/ruxailab/web-eye-tracker/issues/11#issuecomment-1991161128) to avoid duplicates. It would be great if you could chip in. Thanks!

**Reply by KarinePistili** [reply]
Nice! That's great! I'll close this and we can follow up there ;)

**Comment by KarinePistili** [comment]
Closing this, as discussion in continuing on https://github.com/ruxailab/web-eye-tracker/issues/11
